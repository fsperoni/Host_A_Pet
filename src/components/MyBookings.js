import React, { useContext, useEffect, useState } from 'react';
import UserContext from '../hooks/useUserContext';
import HostAPetApi from './Api';
import MyBooking from './MyBooking';


const MyBookings = () => {
  const { currentUser } = useContext(UserContext);
  const [bookings, setBookings] = useState([]);
  const [reviews, setReviews] = useState([]);

  useEffect(function getMyBookings() {
    const getBookings = async () => {
      const userBookings = await HostAPetApi.getUserBookings(currentUser.username);
      setBookings(userBookings);
    }
    getBookings();
}, [currentUser.username]);

useEffect(function getMyReviews() {
  const getReviews = async () => {
    const userReviews = await HostAPetApi.getUserReviews(currentUser.username);
    setReviews(userReviews);
    console.log("All", userReviews);
  }
  getReviews();
}, [currentUser.username]);

const bookingItems = bookings.map(booking => (
  <MyBooking key={booking.id} booking={booking} reviews={reviews} />
))

return (
  <div className="MyBookings mt-3">
    <div className="container">
      {bookingItems}
    </div>
  </div>
)
}

export default MyBookings;