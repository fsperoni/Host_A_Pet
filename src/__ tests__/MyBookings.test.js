import { render } from "@testing-library/react";
import UserContext from '../hooks/useUserContext';
import MyBookings from "../components/MyBookings";

const {
  currentAdminUser,
} = require("./_testCommon");

// smoke test
it('renders without crashing', function () {
  render(
    <UserContext.Provider
      value={{ currentUser: currentAdminUser(), function() { } }}>
      <MyBookings />
    </UserContext.Provider>);

});

//snapshot test
it("matches snapshot", function () {
  const { asFragment } = render(
    <UserContext.Provider
      value={{ currentUser: currentAdminUser(), function() { } }}>
      <MyBookings />
    </UserContext.Provider>);
  expect(asFragment()).toMatchSnapshot();
});
