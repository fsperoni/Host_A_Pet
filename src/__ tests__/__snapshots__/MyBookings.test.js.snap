// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MyBookings mt-3"
  >
    <div
      class="container"
    />
  </div>
</DocumentFragment>
`;
